src/mbox/model.rs

```diff
@@ -33,58 +34,71 @@ impl<'a> ser::Serialize for SerializableAttribute<'a> {
     }
 }
 
+/// Represents the attributes of a mailbox.
 #[derive(Debug, PartialEq)]
-pub struct Attributes<'a>(&'a [imap::types::NameAttribute<'a>]);
-
-impl<'a> From<&'a [imap::types::NameAttribute<'a>]> for Attributes<'a> {
-    fn from(attrs: &'a [imap::types::NameAttribute<'a>]) -> Self {
-        Self(attrs)
+pub struct Attributes(pub HashSet<NameAttribute<'static>>);
+
+impl<'a> From<&[NameAttribute<'a>]> for Attributes {
+    fn from(attrs: &[NameAttribute<'a>]) -> Self {
+        Self(attrs
+                .iter()
+                .map(|attribute| convert_to_static(attribute).unwrap())
+                .collect::<HashSet<NameAttribute<'static>>>()
+        )
     }
 }
 
-impl<'a> ToString for Attributes<'a> {
+impl ToString for Attributes {
     fn to_string(&self) -> String {
-        match self.0.len() {
-            0 => String::new(),
-            1 => {
-                let attr = &SerializableAttribute(&self.0[0]);
-                let attr: &str = attr.into();
-                attr.to_owned()
-            }
-            _ => {
-                let attr = &SerializableAttribute(&self.0[0]);
-                let attr: &str = attr.into();
-                format!("{}, {}", attr, Attributes(&self.0[1..]).to_string())
-            }
+        let mut attributes = String::new();
+
+        for attribute in &self.0 {
+            let attribute = SerializableAttribute(&attribute);
+            attributes.push_str((&attribute).into());
+            attributes.push_str(", ");
         }
+
+        // remove the trailing whitespace with the comma
```

Comment:
What about [`.trim_end_matches()`](https://doc.rust-lang.org/std/string/struct.String.html#method.trim_end_matches)?

Reply:
well it's just a single whitespace, so I thought that pop might suit here. The compiler knows that he needs to remove only one character instead of looking after more whitespaces. At least I think that the compiler will think like this.

Reply:
But what if _one time_ there is no white space? Or if we have an empty `String`? The program will panic, where the `trim_end_matches` will not. I know it's a bit extreme, but the less panicking code we have the better we will be (in my opinion).

Reply:
Hm... ok, hold ok, I'll fix that quickly in a quick PR. (sorry, missread this part... ._.)